src/terra-dev-site/app/components.2/BannerPropsTable.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/* eslint-disable react/no-unused-prop-types */
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the banner.
+   */
+  action: PropTypes.element,
+  /**
+   * The pieces to populate a banner when type=`custom`.
+   */
+  custom: PropTypes.shape({
+    /**
+     * The title for the banner which will be bolded.
+     */
+    bannerTitle: PropTypes.string,
+    /**
+     * The class name used to set the icon in the banner.
+     */
+    icon: PropTypes.element,
+    /**
+     * Sets an author-defined class, to control the status bar color to be used.
+     */
+    colorClass: PropTypes.string,
+  }),
+  /**
+   * Nodes providing the message content for the banner. Can contain text and HTML.
+   */
+  description: PropTypes.node,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause
+   * the Dismiss button to be included on the banner.
+   */
+  onDismiss: PropTypes.func,
```

Comment:
I've been thinking about this class of prop with some of the work I'm doing. Our modal today has a prop called `onRequestClose` that gets called when the modal thinks it should be dismissed. The 'request' part is indicating that the modal won't _actually_ go away until the consumer actually makes it happen.

I believe the same thing is true here; until the consumer of the Banner unmounts it, it's not actually gone. My question is whether we like the 'request' nomenclature for these types of callbacks (and should include it here) or do we feel like it's too wordy and consumers will get the point without it?

Reply:
updated to onRequestClose: https://github.com/cerner/terra-application/pull/45/commits/c36f75662652b37024088bf93d033b73d9d4c71b